### PR TITLE
Avoid auto-fixing UP032 if comments are present around format call arguments

### LINKED ISSUE #6336: Autofix for UP032 swallows comment
Given this snippet:

```py
        return "data:image/{};base64,{}".format(
            ext[1:], data.decode()  # Remove the leading dot.
        )
```

Ruff autofix produces this output:

```py
        return f"data:image/{ext[1:]};base64,{data.decode()}"
```

Note that the autofixed code "swallows" the comment. See here for original context: https://github.com/WhyNotHugo/django-afip/pull/186/commits/2974184b776bede851eba0632b80d915673dea81

This is reproducible with ruff 0.0.281. [The relevant pyproject.toml is here](https://github.com/WhyNotHugo/django-afip/blob/2974184b776bede851eba0632b80d915673dea81/pyproject.toml).

### PR DESCRIPTION
**Pull request by harupy**
Fix #6336 

## Test Plan

A new test case

### CODE REVIEW

File: crates/ruff/src/rules/pyupgrade/rules/f_strings.rs [hunk 1]
```diff
@@ -420,7 +422,19 @@ pub(crate) fn f_strings(
...
+    // "{}".format(
+    //     0,  # 0
+    // )
+    // ```
+    if checker.patch(diagnostic.kind.rule())
+        && !checker
+            .indexer()
+            .comment_ranges()
+            .intersects(TextRange::new(attr.end(), expr.end()))
+    {
```
Comment (zanieb) [inline crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:437]: Alternatively we could downgrade from a `suggested` to a `manual` fix if we detect comments?

Comment (zanieb) [reply crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:437]: (Although this change is probably best for right now)

File: crates/ruff/src/rules/pyupgrade/rules/f_strings.rs [hunk 2]
```diff
@@ -420,7 +422,19 @@ pub(crate) fn f_strings(
...
+    // ```
+    // "{}".format(
+    //     0,  # 0
+    // )
+    // ```
+    if checker.patch(diagnostic.kind.rule())
+        && !checker
+            .indexer()
+            .comment_ranges()
+            .intersects(TextRange::new(attr.end(), expr.end()))
```
Comment (charliermarsh) [inline crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:436]: Instead, can you grab the `arguments` node off the call (line 321), and use `arugments.range()`? To me that better matches the semantics and is a bit more readable.

Comment (harupy) [reply crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:436]: Great suggestion. Yeah, that's much better.

**Review by charliermarsh — APPROVED** [review]

**Pull request merged**